### DISCUSSION TITLE
Option to keep a pane after command exits

### DISCUSSION THREAD

**Discussion by devmatteini** [zellij-org/zellij#2137] [Ideas]
Hi and thanks for this amazing tool! The layout system was exactly what I was looking for my needs!

Given this simple layout file
```
layout {
    pane split_direction="vertical" {
        pane name="main"
        pane name="typecheck" {
            command "npm"
            args "run" "typecheck:w"
        }
    }
}
```
which will render something like this
```
┌───────────────┬──────────────────┐
│  main         │ typecheck        │
├───────────────┼──────────────────┤
│$              │                  │
│               │ watching...      │
│               │                  │
│               │                  │
│               │                  │
│               │                  │
└───────────────┴──────────────────┘
```

When I press CTRL+C on the typecheck pane to stop the watcher, and then again CTRL+C to exit the command, the typecheck pane will be closed.
```
┌─────────────────────────────────────┐
│  main                               │
├─────────────────────────────────────┤
│  $                                  │
│                                     │
│                                     │
│                                     │
└─────────────────────────────────────┘
```

I would like an option to keep the pane position/layout without restarting the command:
```
┌─────────────────┬───────────────────┐
│  main           │typecheck          │
├─────────────────┼───────────────────┤
│  $              │$                  │
│                 │                   │
│                 │                   │
│                 │                   │
└─────────────────┴───────────────────┘
```

Do you think this could be something zellij could support?

**Comment by imsnif** [comment]
Glad you like Zellij!

Seems to me we already do this with the <ENTER> to re-run <CTRL-C> to exit mode though... what do you think would be the benefit of another such layer?

**Reply by devmatteini** [reply]
Mainly to change the command to run and not having to create the pane again manually

**Reply by imsnif** [reply]
That's a cool use-case. I've often wanted to change the command in a command pane as well.

I think I'd prefer this be implemented as something like: `zellij run --pane-id <pane_id> -- new-command`, what do you think?

**Reply by devmatteini** [reply]
Yeah, I like it!

Out of curiosity, why do you prefer the implementation you suggested instead of adding it in the re-run mode? Wouldn't it be easier to use instead of running the `zellij run ...` command?

**Reply by imsnif** [reply]
I just think until Zellij has an internal shell of its own, it'll be much easier to run things from your own shell (with your own completios, etc.)

Would you like to open an issue for this?

**Reply by devmatteini** [reply]
Sure, I'll open an issue when I can

**Reply by devmatteini** [reply]
Hi! I created the issue: https://github.com/zellij-org/zellij/issues/2147
